### PR TITLE
Donut fixes

### LINKED ISSUE #39939: Caks remove filling from jelly-filled donuts that they've frosted
Issue reported from a round on a personal server

Reporting client version: 512

If you lay a jelly-filled donut next to a cak to be frosted, the cak will indeed frost the donut, but will also strip it of its jelly filling.  The item goes from being a jelly-filled donut to a frosted donut, despite the fact that frosted jelly-filled donuts are an item that exists.

Diabolical!  I _knew_ these feline confections were products of the culinary dark arts!

### Reproduction Steps:

1. Obtain unfrosted jelly-filled donut
2. Locate cak
3. Place unfrosted jelly-filled donut on the ground next to cak
4. Wait 1-5 seconds
5. **Problem Behavior:** _The jelly-filled donut becomes a frosted donut, instead of a frosted jelly-filled donut_

**EDIT:** Examining the problem further, I've noticed a couple things.  One; frosted donuts and frosted jelly donuts are not separate items from their normal counterparts, at least if the admin create item menu is to be believed.  Spawning any donut has a chance at random to produce the frosted variant instead of the normal one.

Interestingly, donuts that used to be jelly-filled prior to caks frosting them still have the description of "You jelly?" - which was their description as jelly donuts prior to frosting.  They stack, however, with normal frosted donuts if placed into a food cart, indicating that the game regards them as the same item.  Food carts track jelly donuts and frosted jelly donuts separately.

Examining the variables of jelly donuts that have been converted to unfilled frosted donuts, they actually retain the jelly reagent, and the jelly taste, and are still being defined as obj/item/reagent_containers/food/snacks/donut/jelly.  They're just using the standard frosted donut sprite and name, which seems to be what's causing them to both look like and stack with typical frosted donuts instead of their jelly-filled brethren.

### PR DESCRIPTION
**Pull request by ShizCalev**
Fixes #39939

:cl: ShizCalev
fix: Caks will no longer override the bonus reagents provided in a donut when frosting them.
fix: Caks can no longer create frosted frosted jelly donuts.
fix: Jelly donuts will no longer lose their vitamins when they're frosted.
fix: Fixed chaos donuts potentially doubling the amount of reagents added when microwaved with something else.
fix: Donuts now always contain 1 sprinkles as was stated on the wiki. Frosted donuts have a chance at adding an extra sprinkle.
/:cl:

### CODE REVIEW

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 1]
```diff
@@ -48,18 +54,12 @@
 	. = ..()
 	extra_reagent = pick("nutriment", "capsaicin", "frostoil", "krokodil", "plasma", "cocoa", "slimejelly", "banana", "berryjuice", "omnizine")
 	reagents.add_reagent("[extra_reagent]", 3)
-	bonus_reagents = list("[extra_reagent]" = 3, "sugar" = 1)
-	if(prob(30))
-		icon_state = "donut2"
-		name = "frosted chaos donut"
-		reagents.add_reagent("sprinkles", 2)
-		bonus_reagents = list("sprinkles" = 2, "[extra_reagent]" = 3, "sugar" = 1)
```
Comment (ShizCalev) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: this was already handled by initialize_cooked_food()'s reagent transfer.

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 2]
```diff
@@ -8,21 +8,27 @@
...
+	var/is_frosted = FALSE
+	var/extra_reagent = null
 
 /obj/item/reagent_containers/food/snacks/donut/Initialize()
 	. = ..()
 	if(prob(30))
-		icon_state = "donut2"
-		name = "frosted donut"
-		reagents.add_reagent("sprinkles", 2)
-		bonus_reagents = list("sprinkles" = 2, "sugar" = 1)
```
Comment (ShizCalev) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: again, already handled by `initialize_cooked_food()`

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 3]
```diff
@@ -69,12 +69,6 @@
 	. = ..()
 	if(extra_reagent)
 		reagents.add_reagent("[extra_reagent]", 3)
-	if(prob(30))
-		icon_state = "jdonut2"
-		name = "frosted jelly Donut"
-		reagents.add_reagent("sprinkles", 2)
-		bonus_reagents = list("sprinkles" = 2, "sugar" = 1)
```
Comment (ShizCalev) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: again, already handled by `initialize_cooked_food()`

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 4]
```diff
@@ -284,12 +284,8 @@
 	if(health < maxHealth)
 		adjustBruteLoss(-8) //Fast life regen
 	for(var/obj/item/reagent_containers/food/snacks/donut/D in range(1, src)) //Frosts nearby donuts!
-		if(D.icon_state != "donut2")
-			D.name = "frosted donut"
-			D.icon_state = "donut2"
-			D.reagents.add_reagent("sprinkles", 2)
-			D.bonus_reagents = list("sprinkles" = 2, "sugar" = 1)
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: .

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 5]
```diff
@@ -48,18 +54,12 @@
 	. = ..()
 	extra_reagent = pick("nutriment", "capsaicin", "frostoil", "krokodil", "plasma", "cocoa", "slimejelly", "banana", "berryjuice", "omnizine")
```
Comment (ShizCalev) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: to note, both on live and in this PR, `"banana"` / banana juice reacts with the sugar in the donut and causes it to consume 2 sugar, 2 banana juice, and creates 20u of Laughter instead.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**